eldritch/stories.py

```diff
@@ -0,0 +1,402 @@
+    if isinstance(event, events.DiceRoll) and state.turn_phase == "upkeep" and not self.exhausted:
+      # Note: I see nothing that says it has to be one of her dice
+      choice = events.MultipleChoice(owner, "Choose a die to reroll", event.roll[:])
+      chosen = values.Calculation(choice, "choice_index")
+      return events.Sequence([
+          events.ExhaustAsset(owner, self),
+          choice,
+          events.RerollSpecificDice(event.character, event, chosen),
+      ], owner)
+    return None
+
+  def get_trigger(self, event, owner, state):
+    # Once per turn, not technically "exhaust to ..."
+    if (
+        self.exhausted
+        and isinstance(event, events.Mythos)
+        and event.is_done()
+    ):
+      return events.RefreshAsset(owner, self)
+    return None
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Bless(owner)
+
+
+class NunStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("I Shall Not Want")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Sequence([
+        events.DiscardNamed(owner, "Curse"),
+        events.Bless(owner)
+    ], owner)
+
+
+class NunStory(Story):
+  def __init__(self):
+    super().__init__("He Is My Shepherd", "Fear No Evil", "I Shall Not Want")
+    self.max_tokens["clue"] = 2
+    self.tokens["clue"] = 0
+
+  def get_max_token_event(self, token_type, owner):
+    if token_type == "clue":
+      return self.advance_story(owner, True)
+    return events.Nothing()
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.Bless):
+      return events.AddToken(self, "clue", owner)
+    if (
+        isinstance(event, events.BlessCurse)
+        and (event.character == owner)
+        and event.is_resolved()
+        and values.ItemNameCount(owner, "Curse").value(state) == 1
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class PhotographerStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("There's Your Proof")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.DrawSpecific(owner, "specials", "Retainer")
+
+  def get_interrupt(self, event, owner, state):
+    if (
+        isinstance(event, events.AddToken)
+        and event.character == owner
+        and event.token_type == "must_roll"
+        and event.asset.name == "Retainer"
+    ):
+      return events.CancelEvent(event)
+    return None
+
+
+class PhotographerStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("The Film Is Ruined")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Sequence([
+        events.DiscardNamed(owner, "Retainer"),
+        events.DrawSpecific(owner, "specials", "Bad Credit")
+    ], owner
+    )
+
+  def get_override(self, other, attribute):
+    if isinstance(other, (assets.BankLoan, assets.Retainer)) and attribute == "can_keep":
```

Comment:
What is the difference between cancelling a KeepDrawn and changing the can_keep override? Not suggesting that you change anything, just wondering if they're functionally equivalent.

Reply:
For these cards, I think they are functionally equivalent.  I could imagine a scenario for some other character where they can no longer keep Weapons, but could keep other common/unique items, so if you were drawing multiple you wouldn't want to cancel the whole event. But I don't think there's actually any stories like that, so let's try to keep new mechanics to a minimum.